great_expectations/datasource/fluent/interfaces.py

```diff
@@ -1243,3 +1249,70 @@ def _create_validator(self, *, result_format: ResultFormatUnion) -> V1Validator:
             batch_parameters=self.batch_request.options,
             result_format=result_format,
         )
+
+    def compute_metrics(self, metrics: Metric | list[Metric]) -> MetricResult | list[MetricResult]:
+        if isinstance(metrics, Metric):
+            metrics = [metrics]
+
+        execution_engine = self.data.execution_engine
+        execution_engine.batch_manager.load_batch_list(batch_list=[self])
+        metrics_calculator = MetricsCalculator(execution_engine=execution_engine)
+
+        metrics_calculator_result = metrics_calculator.compute_metrics(
+            metric_configurations=[metric.config for metric in metrics],
+            runtime_configuration=None,
+        )
+        return self._metrics_calculator_result_to_metric_result(metrics_calculator_result)
+
+    @staticmethod
+    def _metrics_calculator_result_to_metric_result(
+        metrics_calculator_result: tuple[_MetricsDict, _AbortedMetricsInfoDict],
+    ) -> MetricResult | list[MetricResult]:
+        raw_metrics = metrics_calculator_result[0]
+        raw_metrics_errors = metrics_calculator_result[1]
+        if len(raw_metrics) + len(raw_metrics_errors) == 1:
+            metric_result_dict = raw_metrics or raw_metrics_errors
+            metric_configuration_id = next(iter(metric_result_dict))
+            value = metric_result_dict[metric_configuration_id]
+            MetricResultType = Batch._get_metric_result_type_from_metric_name(
```

Comment:
style comment: i'd prefer to not access a private member of the class. Can we make this a class method. i believe is also in line with the google style guide recommendation to avoid staticmethod?

Reply:
I may not be up to date about our recent Core conventions. I would typically do this to indicate that this is for internal use only, but I guess we do have `@public_api` for that.